### ISSUE TITLE
Asynchronous select loads the data from previous search term

### ISSUE THREAD

**Issue by joshikeerti** [shentao/vue-multiselect#441]
I am using the async select option of vue-multiselect and I am seeing an issue where the response loaded does not match the search term.
When the user starts typing, the ajax calls are made and the data seems to be picking up the most recent response instead of the reponse matching the current term in the input box. 
So, if the user has typed 'appl' the calls are made for a, app, appl and the response for a comes last, the response for that is picked up.
But the current term in the box is 'appl' and the reponse doesn't match it.
Below are the options that I am using. Is there something that needs change?

<multiselect id="ajax" 
v-model="selectedGroups" 
:options="dsGroups" 
:multiple="true" 
:options-limit="300"
 :limit="3" 
:close-on-select="false" 
:clear-on-select="false" 
:internal-search="false" 
:hide-selected="true"  :searchable="searchable" :preserve-search="false" placeholder="Search for groups" label="groupname" track-by="groupid" @search-change="asyncFind">
 </multiselect>

**Comment by Zip753** [comment]
Hi @joshikeerti.
*tl;dr* Request handling is on you.

While vue-multiselect allows you to specify callback on every change of the input, it is solely your responsibility to handle it properly and it is ultimately up to you how to implement that. Regarding your case, you might consider caching the latest request and aborting it (unless you already got a response) whenever the new data comes in, so that the data from earlier request doesn't ever override your options.

**Comment by SirLamer** [comment]
I use an iterator which increments for each API call, and saved as a constant within each API event. When the API response is returned, before displaying it checks if the one attempting to display is the highest response iterator number to date. This assures display occurs in order. In this way, if you are doing lots of rapid-fire API updates like Google search does, you might still want the "older" responses to show up before the newer ones are available when possible, and this technique allows that. My implementation also performs local filtering on the API-served results so that results appear to be instant when the API is really only updating every half-second or so.
